test/src/code_field/code_controller_readonly_sections_test.dart

```diff
@@ -0,0 +1,95 @@
+import 'package:flutter_code_editor/flutter_code_editor.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import '../common/snippets.dart';
+
+void main() {
+  group('CodeController Read-only Sections', () {
+    test('No sections -> All editable', () {
+      final sectionGroups = <Set<String>>[
+        {},
+        {'nonexistent'},
+      ];
+
+      for (final sections in sectionGroups) {
+        final controller = CodeController(
+          text: TwoMethodsSnippet.full,
+          language: TwoMethodsSnippet.mode,
+          namedSectionParser: const BracketsStartEndNamedSectionParser(),
+          readOnlySectionNames: sections,
+        );
+
+        final readonly = controller.code.lines.lines.map((l) => l.isReadOnly);
+        expect(
+          readonly,
+          List<bool>.filled(controller.code.lines.lines.length, false),
+          reason: '$sections',
+        );
+      }
+    });
+
+    test('Initial, changing', () {
+      int notifiedCount = 0;
+      final controller = CodeController(
+        text: TwoMethodsSnippet.full,
+        language: TwoMethodsSnippet.mode,
+        namedSectionParser: const BracketsStartEndNamedSectionParser(),
+        readOnlySectionNames: {'section1'},
+      );
+      controller.addListener(() {
+        notifiedCount++;
+      });
+
+      final readonly1 = controller.code.lines.lines.map((l) => l.isReadOnly);
+      expect(
+        readonly1,
+        [
+          false,
```

Comment:
Suppose we can make it more clear:
```
          false,  // class MyClass {
          false,  //   void method1() {
          true,   //     if (false) {// [START section1]
          true,   //       return;
          true,   //     }// [END section1]
          false,  //   }
          false,  //
          false,  //   void method2() {// [START section2]
          false,  //     return;
          false,  //   }// [END section2]
          false,  // }
          false,  //
```

Reply:
That would introduce two locations that can fall out of sync and never be auto-detected.